### PR TITLE
integration of x-dash

### PR DESCRIPTION
**Pull request by ludrob**
~~*DO NOT MERGE* => this wont work until the x-dash/x-eventpromo component is available~~
x-eventpromo is now available as pre-release: https://www.npmjs.com/package/@financial-times/x-eventpromo

**Build/run**
You'll need to check out x-dash first.
https://github.com/Financial-Times/x-dash
```
make install
make test
make demo
```

**Changes**
- webpack 4 (not n-webpack)
-  es6 modules, x-dash components
- no compat errors in browser
- fixed all unit tests
- handlebars no longer needed

**X-dash doc**
https://financial-times.github.io/x-dash/packages/x-engine/

### CODE REVIEW

**Comment by Thurst-ft** [comment]
Note to self: 
x-dash needs to be released before this can pass.

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const mappedEvent = mapEventData(eventpromoClientResponse.eventpromo, showVariant);
+
+    xEngine.render(
+        <Eventpromo isPaused="true" {...mappedEvent} />, promoSlotSelector
+    );
+
+	return true;
+}
+
+module.exports = eventPromoInit;
```
Comment (i-like-robots) [inline src/index.js]: This is CommonJS but you're using ESM to import the dependencies

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Comment (i-like-robots) [inline src/index.js]: This is never used

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import xEngine from '@financial-times/x-engine';
```
Comment (i-like-robots) [inline src/index.js]: As mentioned in the channel, x-engine doesn't have any default exports so this should be `{ h, render }`

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		|| !typeof eventpromoClientResponse.eventpromo === 'object'
+		|| Object.keys(eventpromoClientResponse.eventpromo) === []
+	) {
+		throw new Error('no eventpromo match for this event');
+	}
+
+    const mappedEvent = mapEventData(eventpromoClientResponse.eventpromo, showVariant);
+
+    xEngine.render(
+        <Eventpromo isPaused="true" {...mappedEvent} />, promoSlotSelector
```
Comment (i-like-robots) [inline src/index.js]: `isPaused={true}`

File: tests/index.spec.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const {JSDOM} = require('jsdom');
 const fetchMock = require('fetch-mock');
 const config = require('../src/config');
-const eventPromoInit = require('../src/main');
+const eventPromoInit = require('../src/index');
```
Comment (i-like-robots) [inline tests/index.spec.js]: You'll need to point this at your bundled code, x-engine doesn't currently include a Jest plugin (but we do use Webpack's define plugin)

File: src/lib/event-date.js [hunk 6]
```diff
@@ -1,11 +1,10 @@
-
 const oDate = require('o-date');
```
Comment (i-like-robots) [inline src/lib/event-date.js]: Is this getting bundled? If so it should be external

**Comment by ludrob** [comment]
Thanks a lot @i-like-robots.
Thanks to your comments, intellij local history,  some meds and a slightly better working brain, I've managed to get everything to work :) Component renders and tests pass.
I kept on doing and reverting stuff on Wednesday but I had actually done most of the deps cleanup locally at some point. I just kept on struggling with babel.
I've removed preact from the picture for now as well, will try again later.
I'm pretty sure It can be vastly improved, I'm not en expert when it comes to babel/dep management.

> o-date and x-engine are being bundled but will need declared as externals in order for apps to consume the bundle
How do I do this? https://webpack.js.org/configuration/externals/?

**Pull request merged**